ladder/teams/README.md

```diff
@@ -0,0 +1,67 @@
+# How Cilium GitHub Teams Are Managed
+The Cilium project makes use of GitHub teams to help enforce responsibility boundaries with regard to the project codebase here on GitHub.
+GitHub teams loosely map groups of [contributors](https://github.com/cilium/community/blob/main/CONTRIBUTOR-LADDER.md) to the Cilium Project as either reviewers and/or committers, focused on a portion of the codebase. [GitHub teams](https://github.com/orgs/cilium/teams) are an implementation detail in how code access and review responsibilities are organized and are not a 1 to 1 mapping to [Cilium Project teams](https://github.com/cilium/community/blob/main/CONTRIBUTOR-ROLES.md).
+
+For example, the Cilium Project Code team, may require multiple GitHub teams in the Cilium GitHub organization to self-organize their work. 
```

Comment:
In practice the "Cilium Project Code team" is not really something we actively maintain; it's more of a virtual designation for a category of contributors. That role just explains some expectations for some contributors, specifically committers who actively work on code, but we don't track a list of these anywhere or assign special permissions or responsibilities.

Overall I think this doc would probably benefit from just ignoring roles to start off with because roles are for the most part not encoded in these teams.

Reply:
To expand on the above: Roles are designated in the `roles/` directory as markdown files, and aren't part of this directory or any of the corresponding automation.

Reply:
This was added to address @xmulligan suggestion to tie this back to roles language.
Since this directory is in fact under the ladder directory and github has no concept of ladder in its implementation language. There an ambiguity around the word "team" that is both a high level project concept and a github implementation specific concept.  

In effect what we are really doing is using a whole bunch of GitHub subteams, teams within teams, to implement permission boundaries that make sense for our contributor ladder divisions.

Reply:
Sorry, I didn't realize there was review discussion about that.

Reading through the text, I don't think we need an example at this stage of the explanation, this specific sentence about the code team removes clarity rather than adding clarity. I'll post some suggested amended text to tie in the rest.

Reply:
I see "ladder" and "teams" as somewhat orthogonal concepts. There is the general ladder of how we see different contributor's level of involvement in the project. Then for the most part when someone becomes a reviewer or higher on the ladder, they gain the privilege to join a specific team with specific permissions.

I'd probably put the ladder in one directory and teams under a different top-level directory if the directory hierarchy is creating confusion here.